app/controllers/katello/api/rhsm/candlepin_dynflow_proxy_controller.rb

```diff
@@ -29,6 +29,19 @@ def upload_profiles
       render :json => Resources::Candlepin::Consumer.get(@host.subscription_facet.uuid)
     end
 
+    api :PUT, "/systems/:id/deb_package_profile", N_("Update installed deb packages")
+    param :deb_package_profile, Hash, :required => true do
+      param :deb_packages, Array, :required => true do
+        param :name, String, :required => true
+        param :architecture, String, :required => true
+        param :version, String, :required => true
+      end
+    end
+    param :id, String, :desc => N_("UUID of the system"), :required => true
+    def deb_package_profile
+      upload_profiles
+    end
```

Comment:
This is the interface, we want to mark deprecated, right?
Should we not repackage the profile to match the one expected by `upload_profiles`?
Sth like: `profiles = [ {'content_type'=> 'deb', 'profile' => profiles.dig("deb_package_profile", "deb_packages") || []} ]`

Reply:
I will mark this deprecated. I opted to *not* reform the request here because that would require parsing it into JSON and sending it through to the action. It looks like this is avoided by the `upload_profiles` API for performance reasons. I'm OK with leaving it this way for now and changing the Action class implementation once the subscription-manager apt plugin sends the request to the right endpoint and in the right format

Reply:
Marked as deprecated